src/main/java/com/treasuredata/client/TDClientConfig.java

```diff
@@ -0,0 +1,488 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.treasuredata.client;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.common.base.Optional;
+import com.google.common.io.Files;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * TD Client configuration
+ */
+public class TDClientConfig
+{
+    private static Logger logger = LoggerFactory.getLogger(TDClientConfig.class);
+    public static final String ENV_TD_CLIENT_APIKEY = "TD_API_KEY";
+    /**
+     * Keys for configuring TDClient with a properties file (or System properties)
+     */
+    public static final String TD_CLIENT_APIKEY = "td.client.apikey";
+    public static final String TD_CLIENT_USER = "td.client.user";
+    public static final String TD_CLIENT_PASSOWRD = "td.client.password";
+    public static final String TD_CLIENT_USESSL = "td.client.usessl";
+    public static final String TD_CLIENT_API_ENDPOINT = "td.client.endpoint";
+    public static final String TD_CLIENT_API_PORT = "td.client.port";
+    public static final String TD_CLIENT_RETRY_LIMIT = "td.client.retry.limit";
+    public static final String TD_CLIENT_RETRY_INITIAL_WAIT_MILLIS = "td.client.retry.initial-wait";
+    public static final String TD_CLIENT_RETRY_INTERVAL_MILLIS = "td.client.retry.interval";
+    public static final String TD_CLIENT_CONNECT_TIMEOUT_MILLIS = "td.client.connect-timeout";
+    public static final String TD_CLIENT_IDLE_TIMEOUT_MILLIS = "td.client.idle-timeout";
+    public static final String TD_CLIENT_CONNECTION_POOL_SIZE = "td.client.connection-pool-size";
+    public static final String TD_CLIENT_PROXY_HOST = "td.client.proxy.host";
+    public static final String TD_CLIENT_PROXY_PORT = "td.client.proxy.port";
+    public static final String TD_CLIENT_PROXY_USER = "td.client.proxy.user";
+    public static final String TD_CLIENT_PROXY_PASSWORD = "td.client.proxy.password";
+    /**
+     * endpoint URL (e.g., api.treasuredata.com, api-staging.treasuredata.com)
```

Comment:
It's better that 'api-staging' should be changed with 'ybi.jp-east.idcfcloud.com'.